include/RAJA/pattern/kernel/ArgHelper.hpp

```diff
@@ -0,0 +1,179 @@
+/*!
+ ******************************************************************************
+ *
+ * \file
+ *
+ * \brief   Header file for kernel lambda executor.
+ *
+ ******************************************************************************
+ */
+
+
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2016-19, Lawrence Livermore National Security, LLC.
+//
+// Produced at the Lawrence Livermore National Laboratory
+//
+// LLNL-CODE-689114
+//
+// All rights reserved.
+//
+// This file is part of RAJA.
+//
+// For details about use and distribution, please read RAJA/LICENSE.
+//
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+
+#ifndef RAJA_pattern_kernel_arghelper_HPP
+#define RAJA_pattern_kernel_arghelper_HPP
+
+#include "RAJA/config.hpp"
+
+#include <iostream>
+#include <type_traits>
+
+#include "camp/camp.hpp"
+#include "camp/concepts.hpp"
+#include "camp/tuple.hpp"
+
+#include "RAJA/util/macros.hpp"
+#include "RAJA/util/types.hpp"
+
+#include "RAJA/pattern/kernel/Param.hpp"
+#include "RAJA/pattern/kernel/LambdaArgs.hpp"
+
+namespace RAJA
+{
+
+namespace internal
+{
+
+using RAJA::statement::seg_t;
+using RAJA::statement::param_t;
+using RAJA::statement::offset_t;
+using RAJA::statement::LambdaArgs;
+
+//Concatanate lists
+template<typename ListA, typename ListB>
+struct merge_list
+{};
+
+template<typename...itemsA, typename...itemsB>
+struct merge_list<camp::list<itemsA...>, camp::list<itemsB...>>
+{
+  using type = typename camp::list<itemsA...,itemsB...>;
+};
+
+template<typename...itemsA>
+struct merge_list<camp::list<itemsA...>, camp::list<>>
+{
+  using type  = typename camp::list<itemsA... >;
+};
+
+//List Maker
+template<typename Arg>
+struct listMaker
+{
+  using type = typename camp::list<>;
+};
+  
+//Convert LambdaArgs<T, 1, 2, 3> - > camp::list<LambdaArgs<T, 1>, LambdaArgs<T, 2>, LambdaArgs<T, 3> >
+template<typename T, camp::idx_t head, camp::idx_t... tail>
+struct listMaker<LambdaArgs<T, head, tail...>>
+{
+  using type = typename merge_list<camp::list<LambdaArgs<T, head>>,
+              typename listMaker<LambdaArgs<T, tail...> >::type>::type;
+};
+
+//Convert LambdaArgs<T, id> - > camp::list<LambdaArgs<T, id>>
+template<typename T, camp::idx_t id>
+struct listMaker<LambdaArgs<T, id>>
+{
+  using type = typename camp::list<LambdaArgs<T,id>>::type;
+};
+  
+template<typename List>
+struct parser{};
+
+template<>
+struct parser<camp::list<>>
+{
+  using type = camp::list<>;
+};
+
+template <typename Head, typename... Tail>
```

Comment:
Similar change as above for parser... we should really consider adding a `camp::map` for `camp::list`... I'll probably work on that

```cpp
template <typename... Elems>
struct parser<camp::list<Elems...>>
{
  using type = camp::list<typename listMaker<Elems>::type...>;
};
```

Reply:
I don't think we can do the same for the parser since we have the merge list step.

Reply:
Would `camp::transform` and `camp::extend`, essentially map and concat, not work for this?